reactor-netty-core/src/main/java/reactor/netty/FutureMono.java

```diff
@@ -126,20 +113,41 @@ public void subscribe(CoreSubscriber<? super Void> s) {
 				return;
 			}
 
-			if (f.isDone()) {
-				if (f.isSuccess()) {
-					Operators.complete(s);
-				}
-				else {
-					Operators.error(s, FutureSubscription.wrapError(f.cause()));
-				}
-				return;
+			doSubscribe(s, f);
+		}
+	}
+
+	static <F extends Future<Void>> void doSubscribe(CoreSubscriber<? super Void> s, F future) {
```

Comment:
nit: add suppression `@SuppressWarnings("FutureReturnValueIgnored")`

it is related to `addListener`/`removeListener` invocation

Reply:
done